docs/OrderUsage.md

```diff
@@ -0,0 +1,176 @@
+
+#### Notes
+
+- We've created a [postman collection](./commercetools-subscription.postman_collection.json) for the queries that you might use as a playground and test with your use cases easily.
+- This document and queries on it are documented to give some ideas, it's always good to follow the official commercetools documentation for the up to date information.
+- We strongly suggest the Order search API usage for your order queries as it will have better performance on a high number of Orders in a Project.
+  - Queries kept simple to have a compact documentation, so most of the queries do not include the details related to sorting, pagination etc.
+  - On the Order Search API if no sorting is specified, the results are sorted by relevance in descending (desc) order.
+  - The Order Search API does not return the resource data of the matching Orders. Instead, it returns a list of Order IDs, which can then be used to fetch the Orders by their ID.
+
+## How to fetch template-orders ?
+
+You might need to fetch template-orders, to support viewing and editing the template order.
+
+The Order search query example:
+
+```json
+{
+  "query": {
+    "exists": {
+      "field": "custom.schedule",
+      "customType": "StringType"
+    }
+  }
+}
+```
+
+In case you want to filter based on the customer email, then the query might be as below with adding `and` compound expression together with `customerEmail` filter.
+
+```json
+{
+  "query": {
+    "and": [
+      {
+        "exists": {
+          "field": "custom.schedule",
+          "customType": "StringType"
+        }
+      },
+      {
+        "exact": {
+          "field": "customerEmail",
+          "value": "{{customerEmailAddress}}"
+        }
+      }
+    ]
+  }
+}
+```
+
+> Note that the Order Search API does not return the resource data of the matching Orders. Instead, it returns a list of Order IDs, which can then be used to fetch the [Orders by their ID](https://docs.commercetools.com/api/projects/orders#query-orders).
+
+One example order query predicate to fetch this order might be sth like below:
+
+- `id in ("{{orderId1}}","{{orderdId2}}", "{{orderId3}}", ...)`
+
+Query Predicate examples for order API:
+
+- `custom(fields(schedule is defined))`
+- `custom(fields(schedule is defined)) AND customerEmail="{{customerEmailAddress}}"`
+
+### Logs and events related to template orders
+
+1. For order creation, we use [Order import API](https://docs.commercetools.com/api/projects/orders-import#orderimportdraft). According to this an [OrderImportedMessage](https://docs.commercetools.com/api/message-types#orderimportedmessage) will be created by the commercetools platform when a template-order is imported.
+2. In case, template-order **can not be created** from the checkout-order due to non-recoverable errors we log an error. It will contain the error details, together with the draft, and response details from the commercetools API.
```

Comment:
```suggestion
2. In case template-order **can not be created** from the checkout-order due to non-recoverable errors, we log an error. It will contain the error details, together with the draft, and response details from the commercetools API.
```

https://www.grammar-quizzes.com/9-6.html